### DISCUSSION TITLE
Official Flatpak Support?

### DISCUSSION THREAD

**Discussion by MMachado05** [mifi/lossless-cut#1811] [Ideas]
Because flatpaks are very easily universally-supported, easily integrate into software centers (Gnome Software, Discover, etc), and aren't as weirdly vilified as snaps which make some distros make it very difficult to get snap apps (i.e. Mint, Tuxedo), I feel it may be beneficial to have flatpaks be an officially-supported distribution channel, perhaps even as an direct replacement for the snap application. Please let me know your thoughts!

inb4 "There's the AppImage," I'm well aware! However the far more simple maintanability of flatpaks is still, I think, worth something.

**Comment by mifi** [comment]
Flatpaks are not officially supported currently. But there is a [3rd party repo with support](https://github.com/flathub/no.mifi.losslesscut)

**Reply by MMachado05** [reply]
Right, what I was suggesting is whether the flatpak distribution method *could* be officially supported. Perhaps in place of the snap, i.e. the download page for lossless cut would list "Install from Flathub" for it's official Linux distribution method.

I say this because, when the flatpak is unofficially supported like it is, an intermediary has to worry about keeping up to date with upstream updates, and then making sure the flatpak matches them. When this doesn't happen, the flatpak falls behind, like it has currently (flatpak is at version 3.55, whereas the newest release was 3.58) Since the benefit of snap is universality across distros (primarily), why not switch to a distribution method that integrates better with existing software centers? Just my thoughts!

**Reply by mifi** [reply]
Ah, initially I don't think electron-builder supported flatpak (#218), that's why I didn't add any flatpak downloads. But when I look at it again, it now [seems to support flatpak](https://www.electron.build/configuration/flatpak.html). I will try.

**Reply by mifi** [reply]
Hmm, that failed:
```bash
  ⨯ flatpak failed with status code 1  failedTask=build stackTrace=Error: flatpak failed with status code 1
    at ChildProcess.<anonymous> (/home/runner/work/lossless-cut/lossless-cut/node_modules/@malept/flatpak-bundler/index.js:71:16)
    at ChildProcess.emit (node:events:514:28)
    at maybeClose (node:internal/child_process:1105:16)
    at Process.ChildProcess._handle.onexit (node:internal/child_process:305:5)
```
https://github.com/mifi/lossless-cut/actions/runs/7155672372/job/19484549898

not sure what's wrong but I found this: https://github.com/electron/forge/issues/2561

I don't have time to look more into it right now, but I will leave a PR open if someone wants to experiment: #1813

**Reply by KaKi87** [reply]
Hi,
How is this going ?
Thanks

**Reply by mifi** [reply]
i haven't worked on that

**Reply by MMachado05** [reply]
I suppose it could be worth reaching out to those currently maintaining the flatpak? I figure they would have the most relevant experience in building a LosslessCut flatpak, in any case; might just be a matter of officially adopting their deployment pipeline.

**Reply by mifi** [reply]
i'm open to building flatpak if electron-builder has built in support for it (which it seems it has now, and hence I have now enabled flatpak building in the latest nightly). But it looks like the flatpak losslesscut repo https://github.com/flathub/no.mifi.losslesscut is building everything from source, which I would not want to do because i would then have to maintain an extra, separate build system here. It does seem that building everything from source is encouraged, see:  https://github.com/flathub/no.mifi.losslesscut/issues/82#issuecomment-3092046389
but please test the nightly flatpak and see if it works

**Reply by MMachado05** [reply]
The nightly flatpak is coming in at 500MB since it can't leverage the runtimes in Flathub as a remote, but after a very brief test drive, I can say that it's working! I'm sure that the download will be much lighter if it's released to Flathub as a verified package and can leverage other dependencies to drop the final download size.

**Comment by KaKi87** [comment]
Hi,

A few months ago, I started developing a tool named [dynapt](https://git.kaki87.net/KaKi87/dynapt) to automatically fetch updates through APT for packages that don't provide an APT repository (nor an alternative such as Flatpak).

As I've been upgrading LosslessCut through it for quite some time now, I just published my first pre-release of it, which you can download and set up from [this tutorial](https://gist.github.com/KaKi87/f620788e9901abbfef4978eb7ad358b4).

For LosslessCut specifically, replace the following configuration line :

```json
    "apps": []
```

With the following :

```json
    "apps": [
        {
            "name": "LosslessCut",
            "github": {
                "repo": "mifi/lossless-cut"
            },
            "appimage": true
        }
    ]
```

For questions or issues, please submit a comment on the tutorial page.

Thanks

**Comment by mifi** [comment]
have added flatpak building officially now. you can try the latest nightly build

**Reply by MMachado05** [reply]
I would recommend popping into the [Flathub manifest repo](https://github.com/flathub/no.mifi.losslesscut) to see if you can claim it and get the app properly verified on Flathub! I also know @cassidyjames is good with getting apps moved into Flathub officially. (Sorry for the unexpected ping! I just see you in forums like this all the timme and figured this might be your bread and butter.)

**Reply by mifi** [reply]
I've removed the flatpak build again from the official repo, because according to the flathub repo contributors, flathub cannot use externally built flatpaks. I don't know how to proceed exactly, and because my knowledge about flatpak (and my time) is limited, I'm not going to prioritise flatpak

**Reply by MMachado05** [reply]
That's unfortunate, but understandable! Hopefully it becomes a viable path again in the future.
